app/controllers/step_controller.rb

```diff
@@ -9,12 +9,18 @@ def current_tribunal_case
     @current_tribunal_case ||= TribunalCase.find_by_id(session[:tribunal_case_id])
   end
 
-  def update_and_advance(form_class, opts={})
+  def update_and_advance(attr, form_class, opts={})
     hash = params.fetch(form_class.name.underscore, {})
     @form_object = form_class.new(hash.merge(tribunal_case: current_tribunal_case))
     @next_step = params[:next_step].blank? ? nil : params[:next_step]
 
     if @form_object.save
+      # if we are reusing a step (e.g. income, income2, income3)
+      # we need to rename the 'income' attribute to 'income2' (or whatever) for the
+      # DecisionTree#destination call
+      # i.e. { income: "high" } -> { income2: "high" }
+      hash = opts[:as] ? { opts[:as] => hash[attr] } : hash
```

Comment:
A hash comes in like so:

```ruby
{ attribute_name: { ________ } }
```

Most of the time the attribute name will be the same as the decision tree node name. This isn't the case however when we have multiple forms that deal with the same attribute. This gives an "as" option to rename `attribute_name` to something else in order to match what the decision tree expects.

E.g. in https://github.com/ministryofjustice/tax-tribunals-datacapture/pull/17/files#diff-694751162b27096d2ab2e00d1e6f7432R11 – the attribute in the form object is called `what_is_appeal_about`, but the decision tree is expecting it to be called `what_is_appeal_about_challenged` (the name of the form).